modules/auxiliary/gather/vmware_vcenter_vmdir_ldap.rb

```diff
@@ -125,12 +128,62 @@ def pillage(entries)
 
     print_good("Saved LDAP data to #{ldif_filename}")
 
-    policy = entries.find { |entry| entry.dn == policy_dn }
-
-    if policy
+    if (policy = entries.find { |entry| entry.dn == policy_dn })
       print_status('Password and lockout policy:')
       print_line(policy.to_ldif)
     end
+
+    # Process entries with a non-empty userPassword attribute
+    process_hashes(entries.select { |entry| entry[:userpassword].any? })
+  end
+
+  def process_hashes(entries)
+    if entries.empty?
+      print_status('No password hashes found')
+      return
+    end
+
+    entries.each do |entry|
+      # This is the "username"
+      dn = entry.dn
+
+      # https://github.com/vmware/lightwave/blob/637a1935fdd3cae4df6aa8925c69fd5744ab1a88/lwraft/server/middle-layer/password.c#L36-L45
+      type, hash, salt = entry[:userpassword].first.unpack('CH128H32')
+
+      unless type == 1
+        vprint_error("Hash type #{type} is not supported yet (#{dn})")
+        next
+      end
+
+      unless hash.length == 128
+        vprint_error("Hash length is #{hash.length} digits, not 128 (#{dn})")
+        next
+      end
+
+      unless salt.length == 32
+        vprint_error("Salt length is #{salt.length} digits, not 32 (#{dn})")
+        next
+      end
```

Comment:
This will want to be per-type if we add others.

Reply:
My intent was to make it generic using parameters for each type, similarly to vmware approach: https://github.com/vmware/lightwave/blob/637a1935fdd3cae4df6aa8925c69fd5744ab1a88/lwraft/server/middle-layer/password.c#L36-L45

Array of tuples (hash_size, salt_size, hash_func) in python syntax: [ (64, 16, SHA512), (32, 5, SHA256), (32, 0, SHA256), (20, 0, SHA1) ] where the hash_func in the end will by jtr_format name ...

Reply:
That makes sense, but that code isn't implemented yet. We can refactor once it's written.

Reply:
Do you intend to do that in this PR?

Reply:
maybe it's not necessary at the moment, the other types seems to be legacy. we can add later,

Reply:
I've got a refactor to push. Should have better future intent.

Reply:
Pushed. Thank you for your patience.